File: src/test/java/org/openrewrite/python/tree/WhileTest.java
package org.openrewrite.python.tree;

import org.junit.jupiter.api.Test;
import org.openrewrite.test.RewriteTest;

import static org.openrewrite.python.Assertions.python;

public class WhileTest implements RewriteTest {

    @Test
    void whileLoop() {
        rewriteRun(
          python(
            """
              while x:
                  pass
              """
          )
        );
    }
}

File: rewrite/tests/python/all/tree/while_test.py
from rewrite.test import rewrite_run, python


def test_while():
    # language=python
    rewrite_run(
        python(
            """\
            def test(i):
                while i < 6:
                    i += 1
            """
        )
    )


def test_while_else():
    # language=python
    rewrite_run(
        python(
            """\
            def test(i):
                while i < 6:
                    i += 1
                else:
                    i = 10
            """
        )
    )
